### PR TITLE
SantaCache: Add command to print histogram of bucket distribution

### PR DESCRIPTION
**Pull request by russellhancox**
The command only exists in Debug builds.

This currently only prints the distribution of the non-root cache. In the near future I'll unify the caches again which stops this being a problem.

### CODE REVIEW

File: Source/common/SNTKernelCommon.h [hunk 1]
```diff
@@ -99,4 +97,10 @@ typedef struct {
   char pname[MAXPATHLEN];
 } santa_message_t;
 
+// Used for the kSantaUserClientCacheBucketCount request.
+typedef struct {
+  uint16_t per_bucket[1024];
```
Comment (tburgin) [inline Source/common/SNTKernelCommon.h]: Make 1024 a constant and use throughout. Or use `sizeof(counts.per_bucket) / sizeof(uint16_t)` where you are using 1024 throughout.

**Comment by russellhancox** [comment]
PTAL: I've switched from printing the count in every bucket to printing the number of buckets that match a count. For easier reading I've also defaulted to just printing the numbers, unless you pass the -g flag.

File: Source/common/SNTKernelCommon.h (hunk 1 above)
Comment (russellhancox) [reply Source/common/SNTKernelCommon.h]: Done.

File: Source/santactl/Commands/SNTCommandCacheHistogram.m [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          @"  Use -g to get 'graphical' output\n"
+          @"Only available in DEBUG builds.");
+}
+
+- (void)runWithArguments:(NSArray *)arguments {
+  [[self.daemonConn remoteObjectProxy] cacheBucketCount:^(NSArray *counts) {
+    NSMutableDictionary<NSNumber *, NSNumber *> *k = [NSMutableDictionary dictionary];
+    [counts enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+      if (!k[obj]) {
+        k[obj] = @1;
```
Comment (tburgin) [inline Source/santactl/Commands/SNTCommandCacheHistogram.m]: This shouldn't be needed. Your L59 should handle setting 1 as the initial value of k[obj].

**Pull request merged**